component/componenthelper/component.go

```diff
@@ -0,0 +1,62 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package componenthelper // import "go.opentelemetry.io/collector/component/componenthelper"
+
+import (
+	"context"
+
+	"go.uber.org/atomic"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+const (
+	lifecycleShutdown = iota
+	lifecycleRunning
+)
+
+type baseComponent struct {
```

Comment:
Lets! Keen to get this in to help reduce code dup in core and contrib.

Reply:
Here's the PR for reference: https://github.com/open-telemetry/opentelemetry-collector/pull/6893

I agree there is some similarity here, but the needs for each use case are quite different. My use case is specifically for test validation and is concerned with ascertaining the status and history of a component. On the other hand, this PR is concerned with managing the state, but not necessarily exposing it or remembering it. I think it would be preferable not to try to unify these implementations.